### ISSUE TITLE
Unix Launcher won't start the game.

### ISSUE THREAD

**Issue by tastychunks** [muttleyxd/arma3-unix-launcher#248]
Using the AppImage version of the Arma 3 Unix Launcher on steam deck, the game runs fine on Proton 6.3 with -noLaunch enabled but I can't get the game to start through the Unix Launcher. Nothing happens when I click on "start"

**Comment by tastychunks** [comment]
Running the AppImage in Konsole and then trying to start the game prints "[critical] Direct launch failed, exception: Syntax error: VDF: Quote or bracket expected"

**Comment by AndrewDavis** [comment]
I recently got this issue as well. The only thing that I believe has changed since I last launched Arma 3 recently was a Steam update. I believe what has happened is either Steam or Arma 3 in their updates have corrupted the Proton install. I say this because it seems from running the Arma 3 Unix Launcher's command line manually that something goes wrong inside of Proton's files (regarding Steam, I think).

I went and manually deleted all of my Proton-related folders, and I'm having Steam reinstall them to see if that fixes it. Might try a different version of Proton on Arma 3's properties as well.

P.S. For some reason, I had trouble starting Arma 3 Unix Launcher after a restart as well... It seems that running it with `-d` added in the terminal makes that work, at least... Don't know why...

Hope this helps; keep us posted on your own efforts. For my own architecture, see my recent [issue](https://github.com/muttleyxd/arma3-unix-launcher/issues/247).

**Comment by AndrewDavis** [comment]
Okay so some more details (see ***TL;DR Solution*** at the end):

---

An example command line that I would run (with or without `sudo`) is (note that this is after having run Arma 3 once in Steam, so that it creates that Proton `/dist/` folder and so forth):

`[sudo] /home/<user>/.local/share/Steam/steamapps/common/Proton\ 8.0/dist/bin/wine64 "c:\windows\system32\steam.exe" "/home/<user>/.local/share/Steam/steamapps/common/Arma 3/arma3launcher.exe -nolauncher -useBE -exThreads=7 -nologs -noSplash -skipIntro"`

(I pulled the above command from what Arma 3 Unix Launcher appears to be running. I also tried removed the `-` options to no avail.)


I get some output in the terminal that looks like the following (shortened):

```
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
wine: failed to open L"C:\\windows\\system32\\winemenubuilder.exe": c0000135
002c:err:wineboot:process_run_key Error running cmd L"C:\\windows\\system32\\winemenubuilder.exe -a -r" (126).
007c:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0024:err:module:open_builtin_so_file failed to load .so lib "/home/<user>/.local/share/Steam/steamapps/common/
  Proton 8.0/dist/lib64/wine/x86_64-unix/steam.exe.so"
wine: failed to start L"c:\\windows\\system32\\steam.exe"
Application could not be started, or no application associated with the specified file.
ShellExecuteEx failed: Procedure not found.
```

It seems someone has encountered a similar or the same error, regarding the `steam.exe.so` line: https://github.com/Frogging-Family/wine-tkg-git/issues/824


Fortunately for me, Arma 3 appears to be running mostly okay from Steam and the "Windows" launcher in Linux now... However, it doesn't save any settings, no matter what I do. Therefore, I looked at **its** command line, and it does the following:

```
bash /home/<user>/.local/share/Steam/steam.sh -applaunch 107410 -useBE -enableHT -exThreads=7 -noLogs -noSplash -skipIntro

/home/<user>/.local/share/Steam/ubuntu12_32/steam -applaunch 107410 -useBE -enableHT -exThreads=7 -noLogs -noSplash -skipIntro

...(more like this)...
```

So if you just run that first line there from the terminal, adding in `-noLauncher`... you can bypass the need to use Arma 3 Unix Launcher.....!

---
---

***TL;DR: Solution:*** `bash /home/<user>/.local/share/Steam/steam.sh -applaunch 107410 -noLauncher -useBE -enableHT -exThreads=7 -noLogs -noSplash -skipIntro`

**Comment by Freudeheit** [comment]
having the same issue except I'm kinda screwed because the official launcher takes forever to open and is so laggy that its unusable I hope this gets fixed soon

**Comment by AndrewDavis** [comment]
Literally, read my post.

**Comment by Freudeheit** [comment]
The command you sent doesn't work (for me at least)
`bash /home/freudeheit/.local/share/Steam/steam.sh -applaunch 107410 -noLauncher -useBE -enableHT 
steam.sh[230726]: Running Steam on ubuntu 20.04 64-bit
steam.sh[230726]: STEAM_RUNTIME is enabled automatically
steam.sh[230726]: Unpack runtime failed, error code 1
steam.sh[230726]: Error: Couldn't set up the Steam Runtime. Are you running low on disk space?
Continuing...
steam.sh[230726]: Error: You are missing the following 32-bit libraries, and Steam may not run:
libc.so.6
steam.sh[230726]: Can't find 'steam-runtime-check-requirements', continuing anyway
/home/freudeheit/.local/share/Steam/steam.sh: line 792: /home/freudeheit/.local/share/Steam/ubuntu12_32/steam: Too many levels of symbolic links
`
secondly this doesn't account for mods afaik (there's no parameters being passed to launch with any mods) so this just seems like a really roundabout way to launch vanilla arma 3 unless I'm missing something

**Comment by Voskhodyaschiy** [comment]
Switching over to Proton 7.0-6 solved the problem for me.

**Comment by AndrewDavis** [comment]
> The command you sent doesn't work (for me at least) `bash /home/freudeheit/.local/share/Steam/steam.sh -applaunch 107410 -noLauncher -useBE -enableHT steam.sh[230726]: Running Steam on ubuntu 20.04 64-bit steam.sh[230726]: STEAM_RUNTIME is enabled automatically steam.sh[230726]: Unpack runtime failed, error code 1 steam.sh[230726]: Error: Couldn't set up the Steam Runtime. Are you running low on disk space? Continuing... steam.sh[230726]: Error: You are missing the following 32-bit libraries, and Steam may not run: libc.so.6 steam.sh[230726]: Can't find 'steam-runtime-check-requirements', continuing anyway /home/freudeheit/.local/share/Steam/steam.sh: line 792: /home/freudeheit/.local/share/Steam/ubuntu12_32/steam: Too many levels of symbolic links ` secondly this doesn't account for mods afaik (there's no parameters being passed to launch with any mods) so this just seems like a really roundabout way to launch vanilla arma 3 unless I'm missing something

Try with `sudo`, make sure it's using the right paths.

There's a way to do it all with the command line, that's what this app does and what I think the built-in Launcher does. This:
- launches Arma 3 in a way that works when Steam launch does not
- gives you better parameters than a default Steam launch will
- lets you see why the heck the launch is failing
- lets you do mods or whatever you want; check either launcher's output (i.e. Task Manager or equivalent), using Windows or internet if you have to
- also should work in Steam, if you set the launch parameters to the part after `107410`

**Comment by Alexq2222** [comment]
Reverting to version 381 fixes the issue. I've noticed that 383 makes steam think arma 3 is running, perhaps because of steam integration?
